core/string/string_buffer.h

```diff
@@ -118,7 +118,7 @@ StringBuffer<SHORT_BUFFER_SIZE> &StringBuffer<SHORT_BUFFER_SIZE>::append(const c
 template <int SHORT_BUFFER_SIZE>
 StringBuffer<SHORT_BUFFER_SIZE> &StringBuffer<SHORT_BUFFER_SIZE>::reserve(int p_size) {
 	ERR_FAIL_COND_V_MSG(p_size < length(), *this, "reserve() called with a capacity smaller than the current size. This is likely a mistake.");
-	if (p_size <= SHORT_BUFFER_SIZE || p_size <= buffer.size()) {
+	if (p_size <= SHORT_BUFFER_SIZE || p_size <= buffer.length()) {
```

Comment:
Is this correct?
Are we reserving by size or length?

If size, then the `ERR_FAIL` seems incorrect, if length, then the param name is incorrect.

If we _are_ reserving by size (and size is counting the NULL terminator), then there should be no change if `<=` size rather than length?

Reply:
Right, it _does_ reserve by `size`. I'm not sure if that's a good idea, but that's out of scope for this PR. I'll change this one back.

Reply:
Suspect the `ERR_FAIL_CON_V_MSG` should be for `size()` too rather than `length()` (if so it's pre-existing, but might be worth fixing).